### PR TITLE
Update using-environments.rst

### PR DESCRIPTION
**Pull request by gasse**
Few typos and text rewriting. Let me know what you think.

### CODE REVIEW

**Comment by AntoinePrv** [comment]
@dchetelat leaving this review for you :)

**Comment by dchetelat** [comment]
Hi, it looks great! Good job!

There are some places where I felt it was still a bit difficult for beginners to understand, so I took the opportunity to push directly a proposal of changes to the branch, I hope it's ok. Mostly, I changed a bit the explanations in the intro and the details about reset. I also corrected some typos here and there, and made the language a bit more formal in some places where I felt it was a bit too informal. Please feel free to rework on it again until we're all happy. :-)

File: docs/using-environments.rst [hunk 1]
```diff
@@ -120,25 +109,20 @@ The method is parameterized with a problem instance file: the combinatorial
...
 
-* The ``observation`` consists in relevant information extracted from the solver,
-  for the user to decide what the next action will be (typically
-  using a machine learning algorithm).
-* The ``action_set``, if present, describes the set of candidate
-  actions which are valid for the next transition. This allows us to deal with highly dynamic
-  actions sets in a simple way.
+* The ``observation`` consists of information about the state of the solver that should be used to select the next action 
+  to perform (for example, using a machine learning algorithm.)
+* The ``action_set``, when non-null, describes the set of candidate actions which are valid for the next transition.
```
Comment (gasse) [inline docs/using-environments.rst:114]: non-null might be a bit misleading here. What value does the action_set take when it is not used ? None ?

Comment (dchetelat) [reply docs/using-environments.rst:114]: In python, `None`, yes. Maybe we should say that instead

**Pull request merged**